### PR TITLE
move connection code into boto_common

### PR DESCRIPTION
**Pull request by mgwilliams**
This adds a new execution module intended to be used for common boto code. @ryan-lane, please look at this and let me know if this makes sense to you based on our discussions.

The following modules have not been updated (yet):

```
boto_elasticache
boto_elb
boto_iam
boto_rds
boto_route53
boto_secgroup
boto_sns
boto_sqs
```

### CODE REVIEW

**Comment by techhat** [comment]
@mgwilliams what do you think about moving the common code into `salt/utils/boto.py` instead?

**Comment by mgwilliams** [comment]
The rationale for using an execution module was to make the code easily backportable, since ext_utils doesn't (yet) work. There's no compelling reason to backport this particular feature. However, other more needed changes might be made to specific modules. The code should eventually end up in utils.boto, but only once the utils loader has been utilized for sufficient number of releases to make backporting to older releases unnecessary. I do have a personal todo to make `__utils__` available everywhere.

**Comment by mgwilliams** [comment]
regarding the test failures... what is the canonical way to make execution module available from another in unit tests? Is the mock ec2 going to work at all with the connection code moved into another module?

File: salt/modules/boto_common.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+'''
+Common functionality for boto modules
+
+.. versionadded:: TBD
```
Comment (rallytime) [inline salt/modules/boto_common.py]: You can go ahead and through a `Beryllium` in here instead of `TBD`.

**Comment by rallytime** [comment]
@mgwilliams I was worried about these unit tests failing since these modules are heavily mocked. I think we just need to address these with the correct mock values, now that these common functionalities have been moved. I can take a close look at this and submit a PR against your branch fixing these in a day or two, unless you beat me to it.

File: salt/modules/boto_vpc.py [hunk 2]
```diff
@@ -1470,3 +1424,11 @@ def _maybe_set_dns(conn, vpcid, dns_support, dns_hostnames):
     if dns_hostnames:
         conn.modify_vpc_attribute(vpc_id=vpcid, enable_dns_hostnames=dns_hostnames)
         log.debug('DNS hostnames was set to: {0} on vpc {1}'.format(dns_hostnames, vpcid))
+
+
+def _get_conn(region, key, keyid, profile):
```
Comment (rallytime) [inline salt/modules/boto_vpc.py]: It looks like the `key` argument is unused. Is that intended?

**Comment by mgwilliams** [comment]
@rallytime so it seems the tests really don't like my use of an execution module here (since it doesn't have a bash docstring and returns a non-serializable object). After further with @ryan-lane, I'm going to update this to a utils module. However, that requires that the utils module have access to `__context__`, so I"ve opened #22248 to take care of that. I will be updating this PR to utilize the features added in #22248.

**Comment by rallytime** [comment]
@mgwilliams That sounds great. LMK when you've got this updated. I was poking at the tests last week, but didn't get it resolved yet. I can start looking at them again once you've updated this PR.

**Comment by thatch45** [comment]
@mgwilliams what is the relationship here with the utils package you got in? Is that why this is not merging cleanly?

**Comment by mgwilliams** [comment]
Yes. Closing this one.

**Pull request closed without merging**